### PR TITLE
feat(multimodal): support train llava with dummy dataset

### PR DESCRIPTION
**Pull request by Khoray**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: internlm/train/pipeline.py [hunk 1]
```diff
@@ -139,6 +139,14 @@ def _check_module(module):
...
                     setattr(param, IS_WEIGHT_ZERO_PARALLEL, True)
 
+        # for vit and vit project
+        if "vision_tower" in name.lower() or "vision_proj" in name.lower():
+            for param in module.parameters():
+                if gpc.is_initialized(ParallelMode.TENSOR) and is_using_isp():
+                    setattr(param, IS_TENSOR_DATA_PARALLEL, True)
+                elif gpc.is_initialized(ParallelMode.TENSOR) and not is_using_isp():
+                    setattr(param, IS_TENSOR_ZERO_PARALLEL, True)
+
```
Comment (huangting4201) [inline internlm/train/pipeline.py:149]: vision的模型如果都是走tp并行的话，这里应该都是设置IS_TENSOR_ZERO_PARALLEL，不应该有IS_TENSOR_DATA_PARALLEL？

Comment (sunpengsdu) [reply internlm/train/pipeline.py:149]: 这里主要是想兼容后面ISP模式吧？万一以后VIT用ISP了

Comment (JiaoPL) [reply internlm/train/pipeline.py:149]: 目前vit是直接从hf上拉下来的，应该没有tp，这里配置是为了流程走通。我先设置IS_TENSOR_ZERO_PARALLEL吧

Comment (JiaoPL) [reply internlm/train/pipeline.py:149]: 感觉也可以先这么写着，等以后modeling_vit接入公仓？

Comment (huangting4201) [reply internlm/train/pipeline.py:149]: 可以

**Review by sunpengsdu — APPROVED** [review]

**Pull request merged**